plugin/storage/es/spanstore/reader.go

```diff
@@ -337,7 +337,17 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []model.TraceID, st
 		}
 		searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
 		for i, traceID := range traceIDs {
-			query := elastic.NewTermQuery("traceID", traceID.String())
+			// TODO remove in newer versions, added in Jaeger 1.16
+			// read ID without leading zeros
+			// https://github.com/jaegertracing/jaeger/pull/1956 added leading zeros to IDs
+			var legacyTraceID string
+			if traceID.High == 0 {
+				legacyTraceID = fmt.Sprintf("%x", traceID.Low)
+			} else {
+				legacyTraceID = fmt.Sprintf("%x%016x", traceID.High, traceID.Low)
+			}
```

Comment:
I think we only want to do that if the normal ID string starts with a 0, otherwise issue a single query.